discord.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+type DiscordBot struct {
+	Token         string
+	GuildId       string
+	AdminRoleId   string
+	StudentRoleId string
+	ClientId      string
+	ClientSecret  string
+	Db            *sql.DB
+
+	Session *discordgo.Session
+}
+
+var (
+	commands = []*discordgo.ApplicationCommand{
+		// All commands and options must have a description
+		// Commands/options without description will fail the registration
+		// of the command.
+		{
+			Name:        "about",
+			Description: "Lookup information about a discord user who has linked their OSU account",
+			Options: []*discordgo.ApplicationCommandOption{
+				{
+					Type:        discordgo.ApplicationCommandOptionUser,
+					Name:        "user",
+					Description: "User",
+					Required:    true,
+				},
+			},
+		},
+	}
+
+	commandHandlers = map[string]func(b *DiscordBot, i *discordgo.InteractionCreate){
+		"about": func(b *DiscordBot, i *discordgo.InteractionCreate) {
+			if !requireAdmin(b, i) {
+				return
+			}
+			options := i.ApplicationCommandData().Options
+			user := options[0].UserValue(nil)
+
+			row := b.Db.QueryRow(`SELECT name_num, display_name, last_login, student, alum, employee, faculty FROM users WHERE discord_id = ?`, user.ID)
+			var (
+				nameNum     string
+				displayName string
+				lastLogin   int
+				student     bool
+				alum        bool
+				employee    bool
+				faculty     bool
+			)
+			err := row.Scan(&nameNum, &displayName, &lastLogin, &student, &alum, &employee, &faculty)
+			if err != nil {
+				b.Session.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+					Type: discordgo.InteractionResponseChannelMessageWithSource,
+					Data: &discordgo.InteractionResponseData{
+						Content: "User has not linked their OSU account",
+					},
+				})
+				return
+			}
+			content := fmt.Sprintf("**[%s (%s)](<https://www.osu.edu/search/?query=%s>)**\nLast login: <t:%d:f>\n",
+				displayName,
+				nameNum,
+				url.QueryEscape(nameNum),
+				lastLogin,
+			)
+
+			sep := ""
+			if student {
+				content += sep + "Student"
+				sep = ", "
+			}
+			if alum {
+				content += sep + "Alum"
+				sep = ", "
+			}
+			if employee {
+				content += sep + "Employee"
+				sep = ", "
+			}
+			if faculty {
+				content += sep + "Faculty"
+				sep = ", "
+			}
+
+			b.Session.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+				Type: discordgo.InteractionResponseChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData{
+					Content: content,
+				},
+			})
+		},
+	}
+)
+
+func isAdmin(b *DiscordBot, m *discordgo.Member) bool {
```

Comment:
Make these functions on the discord bot. I.e.

```go
func (b *DiscordBot) isAdmin(m *discordgo.Member) bool
```

This applies to the other functions as well